client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/GetRollupJobRequest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.rollup;
+
+import org.elasticsearch.client.Validatable;
+import org.elasticsearch.client.ValidationException;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Request to fetch rollup jobs.
+ */
+public class GetRollupJobRequest implements Validatable {
+    private final String jobId;
+
+    /**
+     * Create a requets .
+     * @param jobId id of the job to return or {@code _all} to return all jobs
+     */
+    public GetRollupJobRequest(final String jobId) {
+        this.jobId = Objects.requireNonNull(jobId, "jobId is required");
```

Comment:
<3 validation here

Reply:
The server-side request (and REST endpoint) converts null, empty or `*` into `_all`.  So we should probably do the same here, or remove that from the server-side request and add the logic to the transport action.

Otherwise an asterisk from the HLRC will go unconverted and the server will try to look for a job literally named `*` :)

Reply:
It looks like REST doesn't support null but the transport client does. Or maybe I'm reading it wrong. I'm not sure we need all of that because we have a "normal" way to query for all.

Reply:
We could totally do a default constructor that sets it to `_all` and ensure this one does not do that behavior w the null check